server/keyshare/myirmaserver/server_email_test.go

```diff
@@ -0,0 +1,57 @@
+//+build !local_tests
+
+package myirmaserver
+
+import (
+	"testing"
+	"time"
+
+	"github.com/privacybydesign/irmago/internal/test"
+)
+
+func TestServerLoginEmail(t *testing.T) {
+	db := &memoryDB{
+		userData: map[string]memoryUserData{
+			"testuser": {
+				id:         15,
+				lastActive: time.Unix(0, 0),
+				email:      []string{"test@test.com"},
+			},
+			"noemail": {
+				id:         17,
+				lastActive: time.Unix(0, 0),
+			},
+		},
+		loginEmailTokens: map[string]string{
+			"testtoken": "test@test.com",
+		},
+		verifyEmailTokens: map[string]int64{
+			"testemailtoken": 15,
+		},
+	}
+	StartMyIrmaServer(t, db, "localhost:1025")
+	defer StopKeyshareServer(t)
```

Comment:
These name differences are quite confusing :)

The `StartMyIrmaServer` also depends on a global again. You could also make it return a handler or a channel to close the server.

Reply:
I've fixed that name and removed the global. Additionally, I have added to that stop function that it also stops the MyIRMA component itself, as opposed to just the HTTP server. I have done all of this for the keyshare server tests as well.